xc/domain.ml

```diff
@@ -260,8 +260,18 @@ let make ~xc ~xs vm_info vcpus domain_config uuid final_uuid =
   let config = {
     ssidref = vm_info.ssidref;
     handle = Uuid.to_string uuid;
-    flags = CDF_IOMMU :: flags;
-    iommu_opts = [];
+    flags = begin match sriov with
+      | true ->
+          debug "VM %s - using CDF_IOMMU" (Uuid.to_string uuid);
```

Comment:
Perhaps we should have a `needs_cdf_iommu` and `needs_ept_disabled`, inside the former check whether there are any pass-through devices assigned (without checking the type, so I think this would look at the assigned PCI devices through PCI_DB?), and the latter only when SRIOV is enabled. Would that be correct?

Reply:
That seems OK, although `s/needs_ept_disabled/needs_sharept_disabled/` to match the flag name.

Reply:
So now we need more cases/flags because  IOMMU_NO_SHAREPT is specific to SRIOV but CDF_IOMMU is not? For now I would CDF_IOMMU unconditional again.

Reply:
As far as I know `CDF_IOMMU` is needed when the domain has a passthrough device assigned. 

IOMMU_NO_SHAREPT is needed when a passthrough device is assigned _and_ the VM can be migrated which is not the case for general PCI passthrough or SRIOV but is the case for NVIDIA vGPU SRIOV.

Reply:
The meaning of these flags is is so specific that we should create a record that passes them down rather than rely on more general flags. I think in general that it rather unfortunate that such a dependency between devices and domains exists in the first place.